### PR TITLE
feat: create account API integration

### PR DESCRIPTION
**Pull request by sss-bct**

### CODE REVIEW

File: context/UIContext.tsx [hunk 1]
```diff
@@ -3,23 +3,34 @@ import { ReactNode, createContext, useState, useContext } from 'react'
...
   toggleLoginDialog: () => void
+  isRegisterDialogOpen: boolean
+  toggleRegisterDialog: () => void
 }
 interface UIContextProviderProps {
   children: ReactNode
 }
 
-export const UIStateContext = createContext({ isLoginDialogOpen: false } as UIContextType)
+export const UIStateContext = createContext({
```
Comment (kibo-kevinwatts) [inline context/UIContext.tsx:13]: This should be renamed to same as file, "UIContext"

File: context/AuthContext.tsx [hunk 2]
```diff
@@ -9,30 +9,36 @@ import {
 } from 'react'
 
 import getConfig from 'next/config'
+import { useRouter } from 'next/router'
 
 import { LoginData } from '@/components/layout/Login/LoginContent/LoginContent'
-import { useUserMutations, useUserQueries } from '@/hooks'
-import { storeClientCookie } from '@/lib/helpers/cookieHelper'
+import { RegisterAccountInputData } from '@/components/layout/RegisterAccount/Content/Content'
```
Comment (kibo-kevinwatts) [inline context/AuthContext.tsx:15]: import type { RegisterAccountInputData

File: context/AuthContext.tsx [hunk 3]
```diff
@@ -9,30 +9,36 @@ import {
...
+import { useUserAccountRegistrationMutations, useUserMutations, useUserQueries } from '@/hooks'
+import { removeClientCookie, storeClientCookie } from '@/lib/helpers/cookieHelper'
 
 import type { CustomerAccount } from '@/lib/gql/types'
 
 interface AuthContextType {
   isAuthenticated: boolean
   user?: CustomerAccount
   login: (params: LoginData, toggleLoginDialog: () => void) => any
+  createAccount: (params: RegisterAccountInputData, toggleRegisterDialog: () => void) => any
```
Comment (kibo-kevinwatts) [inline context/AuthContext.tsx:25]: toggleRegisterDialog  should be `onSuccessCallBack`

File: context/AuthContext.tsx (hunk 2 above)
Comment (sss-bct) [reply context/AuthContext.tsx]: Done.

File: context/AuthContext.tsx [hunk 4]
```diff
@@ -9,30 +9,36 @@ import {
...
-  authError: '',
   user: undefined,
+  authError: '',
   login: () => null,
+  createAccount: () => null,
   setAuthError: () => '',
+  logout: () => '',
 }
 
 export const UserContext = createContext(initialState as AuthContextType)
```
Comment (kibo-kevinwatts) [inline context/AuthContext.tsx:44]: file name differs from context name.
Either change file name to `UserContext` or change all references to be `AuthContext`

**Review by kibo-kevinwatts — APPROVED** [review]

File: hooks/mutations/useUser/useUserAccountRegistrationMutations.spec.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { renderHook } from '@testing-library/react-hooks'
+
+import { registerUserMock } from '../../../__mocks__/stories/userMock'
```
Comment (kibo-chandradeeptalaha) [inline hooks/mutations/useUser/useUserAccountRegistrationMutations.spec.ts:3]: use @

File: hooks/mutations/useUser/useUserAccountRegistrationMutations.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { useMutation, useQueryClient } from 'react-query'
+
+import { CustomerAccountAndAuthInfoInput } from '../../../lib/gql/types'
+import { loginKeys } from '../../../lib/react-query/queryKeys'
```
Comment (kibo-chandradeeptalaha) [inline hooks/mutations/useUser/useUserAccountRegistrationMutations.ts:4]: use @

File: pages/_app.tsx [hunk 7]
```diff
@@ -63,13 +62,13 @@ const App = (props: KiboAppProps) => {
                   categoriesTree={pageProps.categoriesTree || []}
                   sticky={true}
                 />
-                <LoginDialog />
-                <Container maxWidth={'xl'}>
+                <DialogRoot />
+                <Container maxWidth={'lg'}>
```
Comment (kibo-chandradeeptalaha) [inline pages/_app.tsx:66]: Why maxWidth is lg ?

**Review by kibo-chandradeeptalaha — APPROVED** [review]

File: context/ModalContext.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+interface ModalContextType {
+  Component: any
+  props?: any
+  showModal: (showModalProps: ModalStateType) => void
+  closeModal: () => void
+}
+
+interface ModalStateType {
+  Component: any
+  props?: any
```
[the remark this answers is not in the document]
Comment (sss-bct) [reply context/ModalContext.tsx:12]: That is a part of props.

File: hooks/mutations/useUser/useUserAccountRegistrationMutations.spec.ts (hunk 5 above)
Comment (sss-bct) [reply hooks/mutations/useUser/useUserAccountRegistrationMutations.spec.ts:3]: Done.

File: hooks/mutations/useUser/useUserAccountRegistrationMutations.ts (hunk 6 above)
Comment (sss-bct) [reply hooks/mutations/useUser/useUserAccountRegistrationMutations.ts:4]: Done.

File: pages/_app.tsx (hunk 7 above)
Comment (sss-bct) [reply pages/_app.tsx:66]: May be rebase issue, fixed to xl.

File: context/AuthContext.tsx (hunk 4 above)
Comment (sss-bct) [reply context/AuthContext.tsx:44]: Done with AuthContext.

File: context/ModalContext.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+export const ModalContextProvider = ({ children }: ModalContextProviderProps) => {
+  const [modalState, setModalState] = useState<ModalStateType>({
+    Component: null,
+    props: {},
+  })
+
+  const showModal = (showModalProps: ModalStateType) => {
+    setModalState({
+      ...showModalProps,
+      props: { isOpen: true, ...showModalProps.props },
```
Comment (kibo-chandradeeptalaha) [inline context/ModalContext.tsx:35]: I don't think we need isOpen prop to be sent. Remove this

Comment (sss-bct) [reply context/ModalContext.tsx]: Correct. Removed.

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Pull request merged**